inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -895,13 +898,42 @@ void MKLDNNGraphOptimizer::FuseConvolutionAndDWConvolution(MKLDNNGraph &graph) {
                                  everyone_is(1, allPads.end[X_AXIS], allPads.end[Y_AXIS]) &&
                                  everyone_is(1, childLayer->_dilation[X_AXIS], childLayer->_dilation[Y_AXIS]) &&
                                  childLayer->_stride[X_AXIS] == childLayer->_stride[Y_AXIS] &&
-                                 false &&  // TODO [oneDNN]: disabled while not ported
-                                 one_of(childLayer->_stride[X_AXIS], 1 /*, 2*/) &&  // TODO [oneDNN]: stride 2 should also be supported
+                                 withBias &&
+                                 one_of(childLayer->_stride[X_AXIS], 1, 2) &&
                                  childNode->getChildEdgeAt(0)->getDims().ndims() == 4;
 
         return isSupportedParams;
     };
 
+    auto isFusingWorthwhile = [&](MKLDNNNodePtr parentNode, MKLDNNNodePtr childNode) {
+        auto* layer = dynamic_cast<ConvolutionLayer*>(childNode->getCnnLayer().get());
+        if (layer == nullptr)
+            THROW_IE_EXCEPTION << "Cannot get convolution layer " << childNode->getName();
+
+        auto inDims = childNode->inDims[0];
+        auto outDims = childNode->outDims[0];
+         int elemSize = MKLDNNExtensionUtils::sizeOfDataType(MKLDNNExtensionUtils::IEPrecisionToDataType(layer->precision));
+
+         int L3_cache_size = utils::get_cache_size(3, false);
+         int dw_conv_input_size = inDims[0] * inDims[1] * inDims[2] * inDims[3] * elemSize;
+         int dw_conv_output_size = outDims[0] * outDims[1]* outDims[2] * outDims[3] * elemSize;
+
+        auto* parentConvolutionNode = dynamic_cast<MKLDNNConvolutionNode*>(parentNode.get());
```

Comment:
The same as above, it is still possible to use shared pointer through std::dynamic_pointer_cast.

Reply:
changed, thanks